test/deal-investor/post-deal-investors.api-test.ts

```diff
@@ -0,0 +1,309 @@
+      ExpirationDate: PROPERTIES.DEAL_INVESTOR.DEFAULT.expirationDate,
+      IsExpirationDateMaximum: true,
+    };
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequestWithDefaultExpirationDate = requestToCreateDealInvestorInAcbsWithBody(acbsRequestBodyWithDefaultExpirationDate).reply(201);
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyWithoutExpiryDate);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      dealIdentifier,
+    });
+    expect(acbsRequestWithDefaultExpirationDate.isDone()).toBe(true);
+  });
+
+  it('sets the default dealStatus if it is not specified in the request', async () => {
+    const { dealStatus: _removed, ...newDealInvestorWithoutDealStatus } = requestBodyToCreateDealInvestor[0];
+    const requestBodyWithoutDealStatus = [newDealInvestorWithoutDealStatus];
+    const acbsRequestBodyWithDefaultDealStatus = {
+      ...acbsRequestBodyToCreateDealInvestor,
+      DealStatus: {
+        DealStatusCode: PROPERTIES.DEAL_INVESTOR.DEFAULT.dealStatus.dealStatusCode,
+      },
+    };
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequestWithDefaultDealStatus = requestToCreateDealInvestorInAcbsWithBody(acbsRequestBodyWithDefaultDealStatus).reply(201);
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyWithoutDealStatus);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      dealIdentifier,
+    });
+    expect(acbsRequestWithDefaultDealStatus.isDone()).toBe(true);
+  });
+
+  it(`replaces the effectiveDate with today's date if the specified effectiveDate is after today`, async () => {
+    const requestBodyWithFutureEffectiveDate = [{ ...requestBodyToCreateDealInvestor[0], effectiveDate: '9999-01-01' }];
+    const acbsRequestBodyWithTodayEffectiveDate = {
+      ...acbsRequestBodyToCreateDealInvestor,
+      EffectiveDate: new Date().toISOString().split('T')[0] + 'T00:00:00Z',
+    };
+    givenAuthenticationWithTheIdpSucceeds();
+    const acbsRequestWithTodayEffectiveDate = requestToCreateDealInvestorInAcbsWithBody(acbsRequestBodyWithTodayEffectiveDate).reply(201);
+
+    const { status, body } = await api.post(createDealInvestorUrl, requestBodyWithFutureEffectiveDate);
+
+    expect(status).toBe(201);
+    expect(body).toStrictEqual({
+      dealIdentifier,
+    });
+    expect(acbsRequestWithTodayEffectiveDate.isDone()).toBe(true);
+  });
+
+  withStringFieldValidationApiTests({
+    fieldName: 'portfolioIdentifier',
+    minLength: 0,
+    maxLength: 2,
+    required: false,
+    generateFieldValueOfLength: (length: number) => valueGenerator.string({ length }),
+    validRequestBody: requestBodyToCreateDealInvestor,
+    makeRequest: (body) => api.post(createDealInvestorUrl, body),
+    givenAnyRequestBodyWouldSucceed: () => {
+      givenAuthenticationWithTheIdpSucceeds();
+      givenAnyRequestBodyToCreateDealInvestorInAcbsSucceeds();
+    },
+  });
```

Comment:
We'll be able to delete this once we remove this request field

Reply:
Deleted